Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -418,6 +418,14 @@ def root_path(self):
             return None
         return result.stdout.rstrip()
 
+    @property
+    @decorators.Memoize()
+    def common_directory(self):
+        result = run([self.executable(), 'rev-parse', '--git-common-dir'], cwd=self.path, capture_output=True, encoding='utf-8')
+        if result.returncode:
+            return os.path.join(self.root_path, '.git')
```

Comment:
would suggest falling back to `--git-dir` first, if we care about supporting git < 2.5

Reply:
`git` 2.5 came out in 2015. Given that in order to run into issues here you have to both be on `git` < 2.5 and have some configuration that means that your repo root doesn't have a `.git` (the most common would be work trees), I think the extra fallback is not worth it